app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -95,328 +49,146 @@ class TableVisualization extends Visualization {
+            rowData = parsedData.json[0].map((_, i) => parsedData.json.map(r => toRender(r[i])))
+        } else if (parsedData.json !== undefined && isObjectMatrix(parsedData.json)) {
+            let firstKeys = Object.keys(data[0])
+            columnDefs = firstKeys.map(key => ({ field: key }))
+            rowData = parsedData.json.map(obj =>
+                firstKeys.reduce((acc, key) => ({ ...acc, [key]: toRender(obj[key]) }), {})
+            )
+        } else if (parsedData.json !== undefined && Array.isArray(parsedData.json)) {
+            columnDefs = [{ field: 'row' }, { field: 'value' }]
+            rowData = parsedData.json.map((row, i) => ({ row: i + 1, value: toRender(row) }))
         } else if (parsedData.json !== undefined) {
-            const table = genTable(parsedData.json, 0, undefined)
-            tabElem.innerHTML = style + table
+            columnDefs = [{ field: 'value' }]
+            rowData = [{ value: toRender(parsedData.json) }]
         } else {
-            const table = genDataframe(parsedData)
-            let suffix = ''
-            const allRowsCount = parsedData.all_rows_count
-            if (allRowsCount !== undefined) {
-                const includedRowsCount = parsedData.data.length > 0 ? parsedData.data[0].length : 0
-                const hiddenCount = allRowsCount - includedRowsCount
-                if (hiddenCount > 0) {
-                    let rows = 'rows'
-                    if (hiddenCount === 1) {
-                        rows = 'row'
-                    }
-                    suffix =
-                        '<span class="hiddenrows">&#8230; and ' +
-                        hiddenCount +
-                        ' more ' +
-                        rows +
-                        '.</span>'
-                }
-            }
-            tabElem.innerHTML = style + table + suffix
+            const headers = [
+                ...(parsedData.indices_header ? parsedData.indices_header : []),
+                ...parsedData.header,
+            ]
+            columnDefs = headers.map(h => ({ field: h }))
+
+            const rows =
+                parsedData.data && parsedData.data.length > 0
+                    ? parsedData.data[0].length
+                    : parsedData.indices && parsedData.indices.length > 0
+                    ? parsedData.indices[0].length
+                    : 0
+            rowData = Array.apply(null, Array(rows)).map((_, i) => {
+                const row = {}
+                const shift = parsedData.indices ? parsedData.indices.length : 0
+                headers.map(
+                    (h, j) =>
+                        (row[h] = toRender(
+                            j < shift ? parsedData.indices[j][i] : parsedData.data[j - shift][i]
+                        ))
+                )
+                return row
+            })
         }
+
+        const allData = parsedData.all_rows_count !== data.length
```

Comment:
```suggestion
        // If the table contains more rows than an upper limit, the engine will send only some of all rows. If data is truncated, we cannot rely on sorting/filtering, as it will not apply to the whole dataset.
        const isNotTruncated = parsedData.all_rows_count !== data.length
```

Would appreciate the comment and a bit more meaningful variable name to make this easier to understand.